packages/@lwc/engine-core/src/framework/vnodes.ts

```diff
@@ -99,6 +100,7 @@ export interface VBaseElement extends BaseVNode, BaseVParent {
     data: VElementData;
     elm: Element | undefined;
     key: Key;
+    slotAssignment: string | undefined;
```

Comment:
We need to move slot assignment outside the `props` in `ElementData` because static vnodes can also be reassigned during light DOM slot forwarding.

For example:

```html
<x-slot-cmp>
    <div>This is a static element</div>
</x-slot-cmp>
```

```html
<template>
    <slot slot="foo"></slot>
</template>
// x-slot-cmp
```
In the above example, the div, which is a static element needs to be forwarded to the`foo` slot. This can't happen through the `props` because static vnodes don't update attributes from props.